### PR TITLE
webview to agent migration of repo not found check logic + unit tests

### PR DESCRIPTION
**Pull request by eric-j**

### CODE REVIEW

File: shared/agent/src/providers/github.ts [hunk 1]
```diff
@@ -712,22 +670,31 @@ export class GitHubProvider
...
 				const { repos } = SessionContainer.instance();
-				const prRepo = await this.getPullRequestRepo(
-					await repos.get(),
-					response.repository.pullRequest
+				const allRepos = await repos.get();
+				console.warn(
+					"eric response.repository.pullRequest.repository.name",
+					response.repository.pullRequest.repository,
+					allRepos.repos
 				);
```
Comment (dsellarsnr) [inline shared/agent/src/providers/github.ts:678]: Too verbose need to remove or make debug

Comment (eric-j) [reply shared/agent/src/providers/github.ts:678]: Whoops, not supposed to be there

Comment (dsellarsnr) [reply shared/agent/src/providers/github.ts]: There are a few here and there

File: shared/agent/src/providers/thirdPartyIssueProviderBase.ts [hunk 2]
```diff
@@ -112,6 +113,92 @@ export abstract class ThirdPartyIssueProviderBase<
 		return request.description;
 	}
 
+	async getProviderRepo(request: {
+		repoName: string;
+		repoUrl: string;
+		repos: any;
```
Comment (dsellarsnr) [inline shared/agent/src/providers/thirdPartyIssueProviderBase.ts:119]: Should be able to get rid of all the `any` in this function - this one is `CSRepository[]` and gives a type to most of the body - and error looks like it is a `string`

File: shared/ui/store/providerPullRequests/slice.ts [hunk 3]
```diff
@@ -1044,6 +1044,7 @@ export const getProviderPullRequestRepoObject = createSelector(
 	}
 );
 
+// Depreciated, but keeping for now until significant regression tests are made.
 export const getProviderPullRequestRepoObjectCore = (
```
Comment (dsellarsnr) [inline shared/ui/store/providerPullRequests/slice.ts:1048]: I would nuke it, it is in git history. We will never forget about this method.

File: shared/ui/Stream/PullRequests/Bitbucket/PullRequest.tsx [hunk 4]
```diff
@@ -312,6 +312,8 @@ export const PullRequest = () => {
 			textEditorUri: state.editorContext.textEditorUri,
 			reposState: state.repos,
 			checkoutBranch: state.context.pullRequestCheckoutBranch,
+			//not currently in use
+			prRepoId: currentPullRequest?.conversations?.repository?.prRepoId,
```
Comment (dsellarsnr) [inline shared/ui/Stream/PullRequests/Bitbucket/PullRequest.tsx:316]: remove

**Review by dsellarsnr — CHANGES_REQUESTED** [review]
Looking good, just a few small changes.

File: shared/ui/store/providerPullRequests/slice.ts (hunk 3 above)
Comment (bcanzanella) [reply shared/ui/store/providerPullRequests/slice.ts:1048]: +1

File: shared/agent/test/unit/providers/provider.spec.ts [hunk 5]
```diff
@@ -43,4 +58,141 @@ describe("provider", () => {
 			expect(ThirdPartyIssueProvider.supportsCreatingPullRequests(provider)).toEqual(true);
 		});
 	});
+
+	it("gh getProviderRepo url found match", async () => {
```
Comment (bcanzanella) [inline shared/agent/test/unit/providers/provider.spec.ts:62]: huge win here for something that has been a problem for a long while!

**Review by dsellarsnr — APPROVED** [review]

File: shared/ui/src/__tests__/providerPullRequests/reducer.test.ts [hunk 6]
```diff
@@ -1,232 +1,16 @@
-import {
-	FetchThirdPartyPullRequestResponse,
-	GitLabMergeRequest,
-} from "@codestream/protocols/agent";
 import { CSRemote, CSRepository } from "@codestream/protocols/api";
-import { describe, expect, it } from "@jest/globals";
-import { getProviderPullRequestRepoObjectCore } from "../../../store/providerPullRequests/slice";
-
 interface RepoStub extends Omit<Partial<CSRepository>, "remotes"> {
```
Comment (dsellarsnr) [inline shared/ui/src/__tests__/providerPullRequests/reducer.test.ts:2]: Need to remove this whole file to get tests to pass (see vscode in teamcity)

**Pull request merged**